Automattic/wp-calypso
# Gutenberg: add partial revision history support and feature flag

#### Changes proposed in this Pull Request

![screen shot 2018-11-04 at 4 51 21 pm](https://user-images.githubusercontent.com/1270189/47972518-e2fcef00-e051-11e8-98dc-b60fbfac8ec6.png)

This is the first attempt at wiring in the EditorRevisionsDialog , and updates the LastRevision component in the sidebar to trigger the dialog, rather than point to wp-admin post revisions. This also adds a feature flag so we do not render revisions in the sidebar until this is fully functional.

Data management for the original post revisions dialog component is still a bit dirty, and can be improved in follow ups. 

If folks approve this. **Please land**. We want to disable visibility of the revisions for Horizon testing.

If any changes are needed feel free to add them to this branch and 🚢 

**Known Issues**

* Loading Revision history currently via `updatePost` only updates the title rather than title, content and except

cc @aduth @gziolo @youknowriad, any suggestions on a better 'core/editor' action that can do this?

#### Testing instructions

* No regressions to history in /post
![screen shot 2018-11-04 at 4 46 24 pm](https://user-images.githubusercontent.com/1270189/47972442-2e62cd80-e051-11e8-9ed6-855e04ce449c.png)
![screen shot 2018-11-04 at 4 46 37 pm](https://user-images.githubusercontent.com/1270189/47972447-3a4e8f80-e051-11e8-88f8-f5d7da0f24cf.png)

* Using the gutenberg editor at /gutenberg, that has been stickered. Make at least 3 saves.

* They are not visible when starting with `DISABLE_FEATURES=gutenberg/revisions npm start`
![screen shot 2018-11-04 at 4 48 33 pm](https://user-images.githubusercontent.com/1270189/47972474-7aae0d80-e051-11e8-8956-f11fbd6cb2d2.png)

* Revisions are visible on development and wpcalypso. 
![screen shot 2018-11-04 at 4 50 27 pm](https://user-images.githubusercontent.com/1270189/47972509-bf39a900-e051-11e8-8f39-ebb68b212e7e.png)

* Clicking on revisions I see a dialog:
![screen shot 2018-11-04 at 4 51 21 pm](https://user-images.githubusercontent.com/1270189/47972531-0c1d7f80-e052-11e8-9618-02802d6c7494.png)

* Clicking on a revision version, should load the appropriate version
* Clicking on "Load" updates the post title, but not post content, which can be fixed in a follow up.
